libs/core/langchain_core/utils/pydantic.py

```diff
@@ -491,28 +486,30 @@ def _create_root_model_cached(
 
 @lru_cache(maxsize=256)
 def _create_model_cached(
-    __model_name: str,
+    model_name: str,
+    /,
     **field_definitions: Any,
 ) -> type[BaseModel]:
     return _create_model_base(
-        __model_name,
+        model_name,
         __config__=_SchemaConfig,
         **_remap_field_definitions(field_definitions),
     )
 
 
 def create_model(
-    __model_name: str,
-    __module_name: Optional[str] = None,
+    model_name: str,
```

Comment:
we don't want this change I think -- it's a public function and it may conflict with field_definitions

Reply:
The double-underscore is a convention for positional arguments.
Although it's possible to use it non-positionally, it would be quite weird to do so.
The change replaces it with a runtime-checked positional argument thanks to the `/` in the args.
So it is equivalent in the intention and normal usage. But breaking for users who really used it incorrectly.
We can also note that this function is deprecated and `create_model_v2` should be used instead. 
I think it doesn't conflict with `field_definitions` as it concerns positional args, not kwargs.

Reply:
`field_definitions` may have a field called "model_name" I think?

Reply:
I was probably the last one to deal w/ the refactor here -- the concern was working around a bad namespacing API where the positional arguments could collide w/ content in field_definitions over which we don't have control

Reply:
That would be unfortunate but OK. I'll keep the double-underscore then.

Reply:
TIL